### PR TITLE
Completed issue-3-edit-an-entry.

### PR DESCRIPTION
**Pull request by Edward-J-Kim**
https://github.com/Edward-J-Kim/code-journal/assets/126144853/494b2913-762b-4954-8c03-3be7aa6b782c

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -62,6 +80,25 @@ function renderEntry(entry) {
...
+    viewSwap('entry-form');
+
+    for (let i = 0; i < data.entries.length; i++) {
+      if (parseInt(data.entries[i].entryId) === parseInt(clickedEntryId)) {
+        data.editing = data.entries[i];
+      }
+    }
+
+    populateEntryForm();
+  });
```
Comment (shawnkost) [inline js/main.js]: You shouldn't be adding an event listener to the pencil, the event listener should be on the `ul`

File: js/main.js [hunk 2]
```diff
@@ -105,12 +141,81 @@ function viewSwap(view) {
   }
 }
 
+function populateEntryForm() {
+  const titleInput = document.getElementById('title');
+  const imageInput = document.getElementById('image-url');
+  const notesInput = document.getElementById('user-notes');
+
+  const img = document.querySelector('.display .image').firstChild;
```
Comment (shawnkost) [inline js/main.js]: These should be done globally outside the function

File: js/main.js [hunk 3]
```diff
@@ -105,12 +141,81 @@ function viewSwap(view) {
...
+}
+
 const entriesLink = document.querySelector('.menu-entries');
 entriesLink.addEventListener('click', function (event) {
   viewSwap('entries');
 });
 
 const newEntry = document.querySelector('.new-entries');
 newEntry.addEventListener('click', function (event) {
+  const editTitle = document.querySelector('.new-entry');
```
Comment (shawnkost) [inline js/main.js:174]: `querySelector`s should be done outside functions.

File: js/main.js [hunk 4]
```diff
@@ -105,12 +141,81 @@ function viewSwap(view) {
...
+        liToUpdate.replaceWith(updatedEntry);
+      }
+    }
+    data.editing = null;
+    viewSwap('entries');
+  }
+
+  $img.src = './images/placeholder-image-square.jpg';
+  $submitForm.reset();
+});
```
Comment (shawnkost) [inline js/main.js]: You already have this, you don't need this function again

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some changes here Edward!

1. The pencil icon should also not be right next to the text. It should be on the right side of the page. Compare yours to the mockup

File: js/main.js (hunk 4 above)
Comment (Edward-J-Kim) [reply js/main.js]: https://github.com/Edward-J-Kim/code-journal/assets/126144853/d0cbcd4f-75e5-4896-9d37-b085f60e7bde

fixed the following comments.

File: js/main.js [hunk 5]
```diff
@@ -78,14 +90,14 @@ function renderEntry(entry) {
...
+  title.appendChild(notes);
 
   const pencil = document.createElement('i');
   pencil.setAttribute('class', 'fa-solid fa-pencil');
   pencil.setAttribute('data-entry-id', entry.entryId);
-  title.appendChild(pencil);
+  textDivContainer.appendChild(pencil);
 
-  pencil.addEventListener('click', function (event) {
+  ul.addEventListener('click', function (event) {
```
Comment (shawnkost) [inline js/main.js]: This function should be outside the `renderEntry` function

File: js/main.js [hunk 6]
```diff
@@ -157,16 +165,15 @@ function populateEntryForm() {
...
-  }
-}
+// function updateOriginalEntry(entry, entryId) {
+//   const entries = document.getElementsByTagName('ul');
+//   for (const i in entries) {
+//     if (entries[i].attributes && parseInt(entries[i].attributes['data-entry-id'].value) === parseInt(entryId)) {
+//       entries[i].replaceWith(entry);
+//     }
+//   }
+// }
```
Comment (shawnkost) [inline js/main.js]: Delete all old code

**Review by shawnkost — CHANGES_REQUESTED** [review]
Couple changes here Edward! Also one more thing I noticed that I can't leave a comment on because it's from an old branch.

1. Inside your `DOMContentLoaded` function, you should be calling `viewSwap(data.entries)` instead of `viewSwap('entry-form')`

File: js/main.js (hunk 6 above)
Comment (Edward-J-Kim) [reply js/main.js]: updated the codes :)

File: js/main.js [hunk 7]
```diff
@@ -119,7 +119,7 @@ document.addEventListener('DOMContentLoaded', function (event) {
     toggleNoEntries();
   }
 
-  viewSwap('entry-form');
+  viewSwap('data.entries');
```
Comment (shawnkost) [inline js/main.js]: Oops my bad, I actually mean `data.view` & make sure it's not a string

**Review by shawnkost — APPROVED** [review]
Approved, but check out my comment

**Pull request merged**